CREATE_A_ROLE.md

```diff
@@ -1,32 +1,107 @@
 A basic guide for how to create a new role in oasis-roles
 
-Setup environment
-=================
-
-The following Python tools should be installed in either a virtualenv or in your system or elsewise
-should be on your system path:
-
-* ansible
-* ansible-lint
-* yamllint
-* flake8
-* molecule
-
-Create Role
-===========
-
-* `mkdir -p oasis/roles && cd oasis/roles`
-* `git clone git@github.com:oasis-roles/meta_skeleton.git`
-* `ansible-galaxy init --role-skeleton=meta_skeleton my_role_name`
-* `cd my_role_name`
-* Edit README.md, meta/main.yml, and write your code
-
-Setup CI Services
-=================
-
-* Create repo in GitHub
-* Create Travis CI job for new repo
-* Add GH remote and push code
-* `ansible-galaxy import oasis-roles my_role_name` - adds repo to global Galaxy index
-* `ansible-galaxy setup travis oasis-roles my_role_name [Travis CI API key]` - adds auto-updating Galaxy index
-  when Travis CI jobs are completed
+# Setup environment #
+
+* Install [podman](https://podman.io/) using your system package manager.  This
+  example is for Fedora:
+
+  `sudo dnf install podman-docker`
+
+* Create a virtualenv.
+  [virtualenvwrapper](https://virtualenvwrapper.readthedocs.io/en/latest/install.html#basic-installation)
+  is recommended.
+
+  `mkvirtualenv oasis`
+
+* Install Python dependencies.
+
+  `pip3 install ansible tox-ansible pytest-molecule molecule-openstack molecule-podman molecule-docker os-cloud-config`
+
+* Install the OASIS
+  [ansible_collection_system](https://github.com/oasis-roles/ansible_collection_system)
+  collection via Galaxy to use the
+  [molecule_openstack_ci](https://github.com/oasis-roles/ansible_collection_system/tree/master/roles/molecule_openstack_ci)
+  role for OpenStack provisioning.
+
+  `ansible-galaxy collection install git+https://github.com/oasis-roles/ansible_collection_system.git`
+
+* Clone an existing collection.  In this example we'll be using the
+  [ansible_collection_system](https://github.com/oasis-roles/ansible_collection_system)
+  collection.  Git submodules are used to bring in shared repo dependencies.
+
+  `git clone --recurse-submodules git@github.com:oasis-roles/ansible_collection_system.git`
```

Comment:
It might be worth mentioning that, in order to operate properly on these, you need to git clone this within the proper folder structure:

```shell
mkdir -p ansible_collections/oasis_roles
cd ansible_collections/oasis_roles
git clone --recurse-submodules git@github.com:oasis-roles/ansible_collection_system.git system
```

And then to set the value of ANSIBLE_COLLECTIONS_PATH or the collections_path in your ansible.cfg folder to the parent of the `ansible_collections` folder.

Reply:
Doing this would mean running the `molecule_openstack_ci` role from the master branch as well as the tests, correct?  In the past I've just been running that stuff from the default location of `~/.ansible/collections` using the latest released version.  Do we want to run from master when we're just developing a new role?  

Running `molcule_openstack_ci` and the tests from master seems like a separate area of concern compared to new role development.  

I could be off base here since I'm still trying to understand all of the changes to our development process.